polaris-patterns/src/components/PolarisPatternsProvider/PolarisPatternsProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import type {PropsWithChildren} from 'react';
+import React, {createContext, useEffect} from 'react';
+import {I18nextProvider, initReactI18next} from 'react-i18next';
+import {createInstance} from 'i18next';
+
+import {DEFAULT_I18N_DETAILS} from '../../configure';
+
+export const I18nDetails = createContext(DEFAULT_I18N_DETAILS);
+
+const i18next = createInstance({
+  fallbackLng: DEFAULT_I18N_DETAILS.locale,
+  interpolation: {
+    escapeValue: false,
+  },
+  resources: {},
+  returnNull: false,
+});
+
+i18next.use(initReactI18next).init();
+
+interface TranslationDictionary {
+  [key: string]: string | TranslationDictionary;
+}
+
+interface I18nDetails {
+  locale: string;
+  country?: string;
+  currency?: string;
```

Comment:
Perhaps we can call this `currencyCode` to be explicit (and to match the [Shop.currencyCode field](https://shopify.dev/docs/api/admin-graphql/2022-07/objects/Shop#field-shop-currencycode))

```suggestion
  currencyCode?: string;
```